handlers/update.go

```diff
@@ -49,6 +49,7 @@ func MakeUpdateHandler(functionNamespace string, clientset *kubernetes.Clientset
 			//deployment.Spec.Template.Spec.Containers[0].ImagePullPolicy = v1.PullAlways
 
 			deployment.Spec.Template.Spec.Containers[0].Env = buildEnvVars(&request)
+			deployment.Spec.Template.Spec.Containers[0].SecurityContext.ReadOnlyRootFilesystem = &request.ReadOnlyRootFilesystem
```

Comment:
Suggest avoiding nil reference by checking if it exists first, creating if not `SecurityContext`